common/src/main/java/io/smallrye/config/common/utils/StringUtil.java

```diff
@@ -171,9 +171,8 @@ public static String toLowerCaseAndDotted(final String name) {
 
                 // Do not convert to index if the first segment is a number
                 if (beginSegment > 0) {
-                    try {
-                        String segment = sb.substring(beginSegment, i);
-                        Integer.parseInt(segment);
+                    String segment = sb.substring(beginSegment, i);
+                    if (isNumeric(segment)) {
                         sb.replace(beginSegment - 1, beginSegment, "[").append("]");
```

Comment:
Lovely, not crucial, but you can spend a char, not need the string. For the replacement is the same? You can use maybe https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/StringBuilder.html#insert(int,char)

Reply:
There is also a `setChar`, https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/StringBuilder.html#setCharAt(int,char). I've used that instead.